spring-data-mock/src/main/java/com/mmnaseri/utils/spring/data/dsl/mock/KeyGeneratorProvider.java

```diff
@@ -22,15 +27,15 @@
 @SuppressWarnings("WeakerAccess")
 class KeyGeneratorProvider {
 
-    private final Map<Class<? extends Serializable>, List<Class<? extends KeyGenerator>>> generators;
+    private final Map<Class<?>, List<Class<? extends KeyGenerator>>> generators;
 
     KeyGeneratorProvider() {
         final List<Class<? extends KeyGenerator>> discoveredKeyGenerators = getKeyGeneratorTypes();
         generators = new ConcurrentHashMap<>();
         for (Class<? extends KeyGenerator> generatorType : discoveredKeyGenerators) {
             final Class<?> keyType = GenericTypeResolver.resolveTypeArgument(generatorType, KeyGenerator.class);
             assert keyType != null;
-            final Class<? extends Serializable> actualKeyType = keyType.asSubclass(Serializable.class);
+            final Class<?> actualKeyType = keyType.asSubclass(Object.class);
             if (!generators.containsKey(actualKeyType)) {
                 generators.put(actualKeyType, new CopyOnWriteArrayList<Class<? extends KeyGenerator>>());
             }
```

Comment:
```suggestion
            if (!generators.containsKey(keyType)) {
                generators.put(keyType, new CopyOnWriteArrayList<Class<? extends KeyGenerator>>());
            }
```
Also, the line after this, but Github doesn't let me comment on that 🤷‍♂️